### DISCUSSION TITLE
Handling Q/Z on 40% keyboards

### DISCUSSION THREAD

**Discussion by calebdw** [binarybottle/engram-layout#68] [Q&A]
Hello!

I've been trying Engram out for a few days now and love the layout, however, I'm at a lose on how to handle the Q/Z letters.

I'm trying to fit the keymap on a 3X5 Corne while still using the lateral center columns for symbols (I thought about placing Q/Z on the bottom of the center columns, however, I believe there's more valuable / frequently accessed symbols that would better suited for those locations as the Q/Z characters are less common.

Some alternatives that I have considered are creating combos for those two letters (though I'm not sure if combos work with shifting mods in ZMK) or placing them on a different layer. Has anyone used this layout on a smaller 40% keyboard and what are some thoughts?

Thanks!

**Comment by sunaku** [comment]
On a hardware programmable keyboard, you can assign Q and Z to combos or put them on a separate layer to avoid using the right pinky finger for them.  In particular, I have a few ideas for combo placement that I'm going to experiment with to further explore your question:
- index columns: `-?` for Z and `,.` for Q
- lower middle columns: `xj` for Z and `mf` for Q --- this keeps the index finger free so you can use Shift (via home row mods or dedicated pinky shift) to capitalize those combo letters.
- upper middle columns: `yu` for Z and `lw` for Q --- this feels more natural to me and should still work with Shift on the opposite hand for capitalization.
- and so on... :thinking:

**Reply by calebdw** [reply]
@sunaku, the middle column suggestion is very interesting...currently I have the combo on `cg` for `z` and `np` for `q` and use my pinkie fingers to activate by pressing in the middle of the keys (I have choc spaced chocs). But I might look into those other keys!

**Comment by Tristen-Sinanju** [comment]
I ended doing a couple moves with my atreus. Should be a layout of mine on
here for that somevhere

Moved z to b, b to left thumb, and q to right thumb next to space. It works
well enough for me and I'm in the 100wpm range. Took a little while to
learn the thumbs as a letter, but I ended up enjoying it.

I tend to type lower on the keyboard and have short pinkies... So I moved b
so I can reach it easier, and left v because it would overburden my thumb
to have both v and space on it. Q seemed fine, how little it's used

**Reply by calebdw** [reply]
@Tristen-Sinanju, that's interesting moving the letters to the thumb...not sure I'd want to sacrifice the keys that are already there though for the least commonly used letters. I was reading an article that suggested moving a common letter like `t` to the thumb cluster, but it hasn't been something I've played around with

**Reply by Tristen-Sinanju** [reply]
here's what i've been working with. i've thought of putting  t there, but the roll on engram for t works out really well; didn't feel the need to move it. anyway, here's an example of my current arrangement, including the layers 

some of the "layout-analyzer-breaker" layouts move E or T to the thumb, but a haven't found them super comfortable to type with... just make good scores on things like steve-p's fork , aha 

![engram](https://github.com/binarybottle/engram/assets/84834000/6506f730-3b35-4590-a033-34b13b7c1e5b)

**Comment by sunaku** [comment]
An interesting precedent for putting Q and Z on combos, in [the Squeezebox keyboard](https://peterlyons.com/problog/2022/10/squeezebox-keyboard-v2209/):

- when `y` and `o` are pressed together, they produce `z`
- when `d` and `w` are pressed together, they produce `q`

![](https://peterlyons.com/problog/images/2022/squeezebox-v2209-engram.png)

**Reply by binarybottle** [reply]
Macros are always a nice approach, as long as you don't suffer strain from pressing multiple keys simultaneously.

**Reply by calebdw** [reply]
That's interesting, I've currently settled on combos opposite of that: I press `g` + `x` to produce `z`, and `f` + `p` to produce `q` (on the corne layout)

**Comment by sunaku** [comment]
There has been fruitful experimentation and discussion on this topic over in the Glove80 keyboard's Discord community, where we've discovered new placements for ZQ and BV (to address some folks' concerns about upper row pinky reachability), [as documented here](https://sunaku.github.io/engram-keyboard-layout.html#variations). :tipping_hand_man:  My favorite is variant ZQ<sub>b</sub> by @x10an14: it eliminates lateral movement of the right pinky finger, making the layout truly columnar. 💯

>The Z and Q keys overload the right pinky finger asymmetrically in the Engram
layout due to the physical arrangement of traditional row-staggered keyboards.
This is an acceptable compromise because Z and Q are the least frequent letters
in English: although they overload the right pinky finger, it seldom types them.
>
>Similarly, some people dislike reaching up for the B and V keys with their pinky
fingers or even, in some cases, reaching down for the G and P keys too!  They may
like the Engram layout but nevertheless dismiss it due to such reasons.
>
>![baseline for variants](https://sunaku.github.io/engram-keyboard-layout-variant.png)
>
>The following variations of the Engram layout may offer some ideas of alternative
placements for these objectionable keys that one may find more comfortable, or at
least preferable, on ergonomic keyboards [such as my Glove80](https://sunaku.github.io/moergo-glove80-ergonomic-keyboard.html).

## Z and Q

### Variant Z<sub>h</sub>

![Engram-Zh variant](https://sunaku.github.io/engram-keyboard-layout-variant-zh.png)
- Z<sub>h</sub> = Z on home row

### Variant ZQ<sub>l</sub>

![Engram-ZQl variant](https://sunaku.github.io/engram-keyboard-layout-variant-zql.png)
- ZQ<sub>l</sub> = Z and Q on lower row

### Variant ZQ<sub>b</sub>

**Credit:** This variant was [discovered by](
https://discord.com/channels/877392805654306816/1156332285449220096/1183776026434732073
) @x10an14.

![Engram-ZQb variant](https://sunaku.github.io/engram-keyboard-layout-variant-zqb.png)
- ZQ<sub>b</sub> = Z and Q on bottom row

## B and V

### Variant BV<sub>h</sub> ZQ<sub>u</sub>

![Engram-BVh-ZQu variant](https://sunaku.github.io/engram-keyboard-layout-variant-bvh-zqu.png)
- BV<sub>h</sub> = B and V on home row
- ZQ<sub>u</sub> = Z and Q on upper row

### Variant BV<sub>h</sub> ZQ<sub>l</sub>

![Engram-BVh-ZQl variant](https://sunaku.github.io/engram-keyboard-layout-variant-bvh-zql.png)
- BV<sub>h</sub> = B and V on home row
- ZQ<sub>l</sub> = Z and Q on lower row

### Variant BV<sub>h</sub> ZQ<sub>b</sub>

![Engram-BVh-ZQb variant](https://sunaku.github.io/engram-keyboard-layout-variant-bvh-zqb.png)
- BV<sub>h</sub> = B and V on home row
- ZQ<sub>b</sub> = Z and Q on bottom row

### Variant BV<sub>b</sub> ZQ<sub>u</sub>

**Credit:** This variant was [discovered by](
https://discord.com/channels/877392805654306816/1111469812850380831/1183811865508315257
) @rej156.

![Engram-BVb-ZQu variant](https://sunaku.github.io/engram-keyboard-layout-variant-bvb-zqu.png)
- ZQ<sub>u</sub> = Z and Q on upper row
- BV<sub>b</sub> = B and V on bottom row

### Variant BV<sub>b</sub> Z<sub>h</sub>

![Engram-BVb-Zh variant](https://sunaku.github.io/engram-keyboard-layout-variant-bvb-zh.png)
- Z<sub>h</sub> = Z on home row
- BV<sub>b</sub> = B and V on bottom row

### Variant BV<sub>b</sub> ZQ<sub>l</sub>

![Engram-BVb-ZQl variant](https://sunaku.github.io/engram-keyboard-layout-variant-bvb-zql.png)
- ZQ<sub>l</sub> = Z and Q on lower row
- BV<sub>b</sub> = B and V on bottom row
